frontend2/src/app/(home)/CanvasContainer.tsx

```diff
@@ -8,29 +8,27 @@ import { flex } from "styled-system/patterns";
 
 import { Skeleton } from "@/components/Skeleton";
 import { PIXELS_PER_SIDE } from "@/constants/canvas";
+import { APP_CONFIG } from "@/constants/config";
+import { useAptosNetworkState } from "@/contexts/wallet";
 import { isServer } from "@/utils/isServer";
 import { getPixelArrayFromImageElement } from "@/utils/tempCanvas";
 
 import { CanvasOverlay } from "./CanvasOverlay";
 
-const CANVAS_IMAGE_URL = process.env.NEXT_PUBLIC_CANVAS_IMAGE_URL;
-
 export function CanvasContainer() {
   const [canvasContainer, containerBounds] = useMeasure();
+  const { network } = useAptosNetworkState();
```

Comment:
NIT: This will make sure that this component is only subscribed to changes from the network field
```suggestion
  const network = useAptosNetworkState((s) => s.network);
```